compiler/next/lib/frontend/frontend-queries.cpp

```diff
@@ -142,13 +145,10 @@ const LocationsMap& fileLocations(Context* context, UniqueString path) {
   // Get the result of parsing
   const uast::Builder::Result& p = parseFile(context, path);
   // Create a map of ast to Location
-  std::unordered_map<const ASTNode*, Location> result;
-  for (auto pair : p.locations) {
-    const ASTNode* ast = pair.first;
-    Location loc = pair.second;
-    if (ast != nullptr) {
-      result.insert({ast, loc});
-    }
+  std::unordered_map<const ASTNode*, Location> result(p.locations.size());
+  for (auto& pair : p.locations) {
+    if (pair.first != nullptr)
+      result.insert(pair);
```

Comment:
I thought it was shorter and to-the-point rather than extracting and then re-assembling the pair.

Maybe a comment on the insert line would help: `// Insert { ASTNode, Location } into result`.

Also note the other minor change of setting the bucket size of result to the number of nodes, to lessen the chances of collisions and/or "rebucketing".

Reply:
Yep, it's fine, we'll leave it in.